lib/sidekiq/cron/job.rb

```diff
@@ -13,18 +13,17 @@ class Job
       LAST_ENQUEUE_TIME_FORMAT = '%Y-%m-%d %H:%M:%S %z'
 
       # Use the exists? method if we're on a newer version of Redis.
-      REDIS_EXISTS_METHOD = Gem.loaded_specs['redis'].version < Gem::Version.new('4.2') ? :exists : :exists?
+      REDIS_EXISTS_METHOD = Gem::Version.new(Sidekiq::VERSION) >= Gem::Version.new("7.0.0") || Gem.loaded_specs['redis'].version < Gem::Version.new('4.2') ? :exists : :exists?
 
       # Crucial part of whole enqueuing job.
       def should_enque? time
-        enqueue = false
         enqueue = Sidekiq.redis do |conn|
           status == "enabled" &&
             not_past_scheduled_time?(time) &&
             not_enqueued_after?(time) &&
-            conn.zadd(job_enqueued_key, formated_enqueue_time(time), formated_last_time(time))
+            conn.zadd(job_enqueued_key, formatted_enqueue_time(time), formatted_last_time(time))
```

Comment:
According to the [documentation](https://github.com/mperham/sidekiq/blob/main/docs/7.0-Upgrade.md#redis-client) `Sidekiq.redis` is now returning a `RedisClient`
`zadd` is not part of the api of [redis-client](https://github.com/redis-rb/redis-client)

Reply:
Also https://github.com/mperham/sidekiq/blob/main/docs/capsule.md#redis-pools
> Sidekiq components and add-ons should avoid using Sidekiq.redis or Sidekiq.logger. Instead use the implicit redis or logger methods available on Sidekiq::Component, Sidekiq::Capsule or Sidekiq::{Client,Server}Middleware. Jobs may continue to use Sidekiq.redis, they will automagically use the capsule's pool of concurrency connections.

Reply:
`Sidekiq.redis` returns a `RedisClientAdapter` which handles mapping methods in a way that's generally compatible with previous redis library: https://github.com/mperham/sidekiq/blob/main/lib/sidekiq/redis_client_adapter.rb

I'll look into creating a helper method for avoiding `Sidekiq.redis` 7 onward

Reply:
Reading over, there'll be more issues than simply updating how redis is invoked. `Sidekiq::Options` as a whole is incorrectly implemented since it goes to `Sidekiq.default_configuration` rather than the context specific config

Leaning towards that conversion being better as a follow up PR, potentially when sidekiq-cron drops support for older versions of sidekiq